### PR TITLE
Add :auto_updates stanza for Casks

### PR DESCRIPTION
**Pull request by nickpellant**
I've added a new `:auto_updates` stanza as per the discussion in #13201. This stanza can be overridden using --force if desired.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
Thank you for working on this @nickpellant. @jawshooah, I trust you entirely on the review.

**Comment by nickpellant** [comment]
@vitorgalvao My pleasure. Going to try and take a pass at the new `:required_location` stanza tomorrow if this goes through.

**Comment by nickpellant** [comment]
@vitorgalvao @jawshooah Appreciate you're both busy people and maintaining this project takes a lot of time! I was just wondering what the process/timeline for reviewing & merging core PRs in is for homebrew-cask? If there are some alterations or additions that need to be made for this I'm happy to tinker on it to get it pushed through.

**Comment by jawshooah** [comment]
@nickpellant There's no predefined timeline for reviewing and merging PRs; it comes down to whenever maintainers have the time.

That said, I'll try to go over this at some point tonight.

**Comment by nickpellant** [comment]
@jawshooah Great thanks! Look forward to seeing your comments.

**Comment by nickpellant** [comment]
@vitorgalvao @jawshooah Any movement on getting this PR looked at? Happy to make some adjustments if necessary.

**Comment by jawshooah** [comment]
Hi @nickpellant, sorry for the lack of movement on this. My personal and professional lives have been demanding more of me than usual lately, so my OSS activity has been limited to what I can take care of within a few minutes. I promise this is on my radar, and I will get to it as soon as I can.

File: lib/hbc/exceptions.rb [hunk 1]
```diff
@@ -44,6 +44,17 @@ def to_s
...
 end
 
+class Hbc::CaskAutoUpdatesError < Hbc::CaskError
+  attr_reader :token
+  def initialize(token)
+    @token = token
+  end
+
+  def to_s
+    %Q{A Cask for #{token} is already installed and uses auto-updates. Add the "--force" option to force re-install.}
```
Comment (fanquake) [inline lib/hbc/exceptions.rb]: s/uses/using

**Comment by nickpellant** [comment]
@jawshooah Don't worry! I too as evidenced by my delay on getting this change done know all to well how life intervenes. ;)  Hopefully this change allows this PR to move forward @fanquake?

**Comment by fanquake** [comment]
@nickpellant I'll start looking over it.

File: lib/hbc/installer.rb [hunk 2]
```diff
@@ -52,6 +52,11 @@ def self.capture_output(&block)
 
   def install(force=false, skip_cask_deps=false)
     odebug "Hbc::Installer.install"
+
+    if @cask.installed? && @cask.auto_updates && !force
+      raise Hbc::CaskAutoUpdatesError.new(@cask)
```
Comment (jawshooah) [inline lib/hbc/installer.rb]: I don't see the need for this new error type. `CaskAlreadyInstalledError` works fine here.

**Comment by jawshooah** [comment]
Apart from my comment above, this looks fine to me. Could use a rebase since it's been a while.

**Comment by vitorgalvao** [comment]
https://github.com/caskroom/homebrew-cask/pull/16102.

**Pull request closed without merging**